File: src/entities/Suivi.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package entities;

import java.util.Date;

/**
 *
 * @author cyrine belhssan
 */
public class Suivi {
     private int idSuivi;
    private int idUser;
     private Personne p;
    private String dateSuivi;
    private float taille;
    private float poid;
  //  private float bmi;
    private float glycemie;
    private String petitDej;
    private String repas;
    private String diner;
    private String notes;

    public int getIdUser() {
        return idUser;
    }

    public void setIdUser(int idUser) {
        this.idUser = idUser;
    }

    public Suivi() {
    }

    public Suivi(int idSuivi, float taille, float poid, float glycemie, String petitDej, String repas, String diner, String notes) {
        this.idSuivi = idSuivi;
        this.taille = taille;
        this.poid = poid;
        this.glycemie = glycemie;
        this.petitDej = petitDej;
        this.repas = repas;
        this.diner = diner;
        this.notes = notes;
    }

    public Suivi(int idSuivi, String dateSuivi, float taille, float poid, float glycemie, String petitDej, String repas, String diner, String notes) {
        this.idSuivi = idSuivi;
        this.dateSuivi = dateSuivi;
        this.taille = taille;
        this.poid = poid;
        this.glycemie = glycemie;
        this.petitDej = petitDej;
        this.repas = repas;
        this.diner = diner;
        this.notes = notes;
    }
    

    public Suivi(String dateSuivi, float taille, float poid, float glycemie, String petitDej, String repas, String diner, String notes) {
        this.dateSuivi = dateSuivi;
        this.taille = taille;
        this.poid = poid;
        this.glycemie = glycemie;
        this.petitDej = petitDej;
        this.repas = repas;
        this.diner = diner;
        this.notes = notes;
    }
    

    public Suivi(int idSuivi, Personne p, String dateSuivi, float taille, float poid, float glycemie) {
        this.idSuivi = idSuivi;
        this.p = p;
        this.dateSuivi = dateSuivi;
        this.taille = taille;
        this.poid = poid;
        this.glycemie = glycemie;
    }
    

    public Suivi(float taille, float poid, float glycemie, String petitDej, String repas, String diner, String notes) {
        this.taille = taille;
        this.poid = poid;
        this.glycemie = glycemie;
        this.petitDej = petitDej;
        this.repas = repas;
        this.diner = diner;
        this.notes = notes;
    }
    
    

    public Suivi(int idSuivi, Personne p, String dateSuivi, float taille, float poid, float glycemie, String petitDej, String repas, String diner, String notes) {
        this.idSuivi = idSuivi;
        this.p = p;
        this.dateSuivi = dateSuivi;
        this.taille = taille;
        this.poid = poid;
        this.glycemie = glycemie;
        this.petitDej = petitDej;
        this.repas = repas;
        this.diner = diner;
        this.notes = notes;
    }

    public Suivi(Personne p, String dateSuivi, float taille, float poid, float glycemie, String petitDej, String repas, String diner, String notes) {
        this.p = p;
        this.dateSuivi = dateSuivi;
        this.taille = taille;
        this.poid = poid;
        this.glycemie = glycemie;
        this.petitDej = petitDej;
        this.repas = repas;
        this.diner = diner;
        this.notes = notes;
    }

    public int getIdSuivi() {
        return idSuivi;
    }

    public Personne getP() {
        return p;
    }

    public String getDateSuivi() {
        return dateSuivi;
    }

    public float getTaille() {
        return taille;
    }

    public float getPoid() {
        return poid;
    }

    public float getGlycemie() {
        return glycemie;
    }

    public String getPetitDej() {
        return petitDej;
    }

    public String getRepas() {
        return repas;
    }

    public String getDiner() {
        return diner;
    }

    public String getNotes() {
        return notes;
    }

    public void setIdSuivi(int idSuivi) {
        this.idSuivi = idSuivi;
    }

    public void setP(Personne p) {
        this.p = p;
    }

    public void setDateSuivi(String dateSuivi) {
        this.dateSuivi = dateSuivi;
    }

    public void setTaille(float taille) {
        this.taille = taille;
    }

    public void setPoid(float poid) {
        this.poid = poid;
    }

    public void setGlycemie(float glycemie) {
        this.glycemie = glycemie;
    }

    public void setPetitDej(String petitDej) {
        this.petitDej = petitDej;
    }

    public void setRepas(String repas) {
        this.repas = repas;
    }

    public void setDiner(String diner) {
        this.diner = diner;
    }

    public void setNotes(String notes) {
        this.notes = notes;
    }

    @Override
    public String toString() {
        return "Suivi{" + "idSuivi=" + idSuivi + ", p=" + p + ", dateSuivi=" + dateSuivi + ", taille=" + taille + ", poid=" + poid + ", glycemie=" + glycemie + ", petitDej=" + petitDej + ", repas=" + repas + ", diner=" + diner + ", notes=" + notes + '}';
    }

   

   
    
}

File: src/Entity/Suivi.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * Suivi
 *
 * @ORM\Table(name="suivi", indexes={@ORM\Index(name="idUser", columns={"idUser"})})
 * @ORM\Entity
 */
class Suivi
{
    /**
     * @var int
     *
     * @ORM\Column(name="idSuivi", type="integer", nullable=false)
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $idsuivi;

    /**
     * @var int|null
     *
     * @ORM\Column(name="dateSuivi", type="integer", nullable=true)
     */
    private $datesuivi;

    /**
     * @var float|null
     *
     * @ORM\Column(name="taille", type="float", precision=10, scale=0, nullable=true)
     */
    private $taille;

    /**
     * @var float|null
     *
     * @ORM\Column(name="poid", type="float", precision=10, scale=0, nullable=true)
     */
    private $poid;

    /**
     * @var float|null
     *
     * @ORM\Column(name="glycemie", type="float", precision=10, scale=0, nullable=true)
     */
    private $glycemie;

    /**
     * @var string|null
     *
     * @ORM\Column(name="petitDej", type="string", length=255, nullable=true)
     */
    private $petitdej;

    /**
     * @var string|null
     *
     * @ORM\Column(name="repas", type="string", length=255, nullable=true)
     */
    private $repas;

    /**
     * @var string|null
     *
     * @ORM\Column(name="diner", type="string", length=255, nullable=true)
     */
    private $diner;

    /**
     * @var string|null
     *
     * @ORM\Column(name="notes", type="string", length=255, nullable=true)
     */
    private $notes;

    /**
     * @var \Personne
     *
     * @ORM\ManyToOne(targetEntity="Personne")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="idUser", referencedColumnName="id")
     * })
     */
    private $iduser;

    public function getIdsuivi(): ?int
    {
        return $this->idsuivi;
    }

    public function getDatesuivi(): ?int
    {
        return $this->datesuivi;
    }

    public function setDatesuivi(?int $datesuivi): self
    {
        $this->datesuivi = $datesuivi;

        return $this;
    }

    public function getTaille(): ?float
    {
        return $this->taille;
    }

    public function setTaille(?float $taille): self
    {
        $this->taille = $taille;

        return $this;
    }

    public function getPoid(): ?float
    {
        return $this->poid;
    }

    public function setPoid(?float $poid): self
    {
        $this->poid = $poid;

        return $this;
    }

    public function getGlycemie(): ?float
    {
        return $this->glycemie;
    }

    public function setGlycemie(?float $glycemie): self
    {
        $this->glycemie = $glycemie;

        return $this;
    }

    public function getPetitdej(): ?string
    {
        return $this->petitdej;
    }

    public function setPetitdej(?string $petitdej): self
    {
        $this->petitdej = $petitdej;

        return $this;
    }

    public function getRepas(): ?string
    {
        return $this->repas;
    }

    public function setRepas(?string $repas): self
    {
        $this->repas = $repas;

        return $this;
    }

    public function getDiner(): ?string
    {
        return $this->diner;
    }

    public function setDiner(?string $diner): self
    {
        $this->diner = $diner;

        return $this;
    }

    public function getNotes(): ?string
    {
        return $this->notes;
    }

    public function setNotes(?string $notes): self
    {
        $this->notes = $notes;

        return $this;
    }

    public function getIduser(): ?Personne
    {
        return $this->iduser;
    }

    public function setIduser(?Personne $iduser): self
    {
        $this->iduser = $iduser;

        return $this;
    }


}
